doc/sphinx/installation.rst

```diff
@@ -114,52 +114,96 @@ ROCm SDK to make use of GPU computation:
     sudo apt update
     sudo apt install libnuma-dev rocm-dkms rocblas rocfft rocrand rocthrust
 
+After installing the ROCm SDK, please reboot your computer.
+
 .. _Installing Requirements on Mac OS X:
 
 Installing Requirements on Mac OS X
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-To make |es| run on Mac OS X 10.9 or higher, its dependencies can be
-installed using MacPorts. First, download the installer package
-appropriate for your Mac OS X version from
-https://www.macports.org/install.php and install it. Then, run the
+Preparation
+"""""""""""
+
+To make |es| run on Mac OS X 10.9 or higher, you need to install its
+dependencies. There are two possibilities for this, MacPorts and Homebrew.
+We recommend MacPorts, but if you already have Homebrew installed, you can use
+that too. To check whether you already have one or the other installed, run the
 following commands:
 
+.. code-block:: bash
+
+    test -e /opt/local/bin/port && echo "MacPorts is installed"
+    test -e /usr/local/bin/brew && echo "Homebrew is installed"
+
+If both are installed, you need to remove one of the two. To do that, run one
+of the following two commands:
+
+.. code-block:: bash
+
+    sudo port -f uninstall installed && rm -r /opt/local
+    ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/uninstall)"
+
+If Homebrew is already installed, you should resolve any problems reported by
+the command
+
+.. code-block:: bash
+
+    brew doctor
+
+If Anaconda Python or the Python from www.python.org are installed, you
+will likely not be able to run |es|. Therefore, please uninstall them
+using the following commands:
+
+.. code-block:: bash
+
+    sudo rm -r ~/anaconda[23]
+    sudo rm -r /Library/Python
+
+If you want to install MacPorts, download the installer package
+appropriate for your Mac OS X version from
+https://www.macports.org/install.php and install it.
+
+If you want to install Homebrew, use the following commands.
+
+.. code-block:: bash
+
+    sudo xcode-select --install
+    sudo xcodebuild -license accept
+    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+
+Installing Packages using MacPorts
+""""""""""""""""""""""""""""""""""
+    
+Run thefollowing commands:
+
 .. code-block:: bash
 
     sudo xcode-select --install
     sudo xcodebuild -license accept
     sudo port selfupdate
     sudo port install cmake python37 py37-cython py37-numpy \
       openmpi-default fftw-3 +openmpi boost +openmpi +python37 \
-      doxygen py37-opengl py37-sphinx py37-pip gsl hdf5 +openmpi
+      doxygen py37-opengl py37-sphinx py37-pip gsl hdf5 +openmpi \
```

Comment:
I think we should put all python dependencies into ```/requirements.txt``` and let the users install them via ```pip install -r requirements.txt```.
That would mean to add the following packages to ```/requirements.txt```:
  * Cython
  * PyOpenGL
  * numpy
  * sphinx
  * matplotlib
  * ipython
  * jupyter

Reply:
I don't agree. If the packages are in the system package manager, they should be installed from there.

Reply:
we have to put the python dependencies there anyway and we are able to restrict the versions if any has a bug

Reply:
> If the packages are in the system package manager, they should be installed from there

why?

Reply:
> . I don't even have pip installed on my laptop because everything is available in MacPorts

we explicitly tell the user to install pip

Reply:
For what it's worth, even on Linux (where many packages are available from the system) I almost always use a venv with the packages for espresso as I find it simpler (also, no changes because of system updates). For execution only (no matplotlib etc), I install always install numpy, cython, and h5py in the venv.

Reply:
If you install other packages from the package manager that require different versions (i.e. the versions provided by the package manager), you'll run into trouble. The only way around is with virtualenv, but we definitely shouldn't confuse people with that. If we want to blacklist versions, we should do that on the CMake or Python level.

Reply:
can you come up with an example where a package from the package manager has a python package dependency if itself is not a python package?

Reply:
Removed pip references as discussed offline with @KaiSzuttor.